### DISCUSSION TITLE
WalletConnect v1/v2 + TrustWallet: Can not send transaction

### DISCUSSION THREAD

**Discussion by liringlas** [wevm/wagmi#1983] [Q&A] [answered]
Greetings! 
While having overall good experience when connecting WalletConnect with TrustWallet, I can not proceed with sending transaction (trying to send USDC/USDT tokens).
Tried it both on WalletConnect v1 and v2.
When trying on v1, TrustWallet throws **"Unknown Error has Occured"** on the screen.
When trying on v2 ( also can be reproduced using live example in here https://wagmi.sh/examples/send-transaction ) -- nothing happens at all.
Using Ethereum mainnet, not test networks.
If anyone had similar issues and came up with any kind of resolution -- advice would be appreciated.
Thanks in advance.

**Comment by deantek** [comment]
same problem, no one answers here as if the problem does not exist

**Comment by liringlas** [comment, marked as answer]
Hello @deantek ! I got my issue resolved and just forgot to write solution back here.
I had a chat with a person from TrustWallet and as it happened the issue was using `gasLimit: BigNumber` field value from `PrepareWriteContractResult` object made by `wagmi`.
So what I've done is I converted `gasLimit: BigNumber` to `gasLimit: string` by doing `String(gasLimit.toNumber())`.
Then it worked.
Though you may need to disconnect/connect TrustWallet to your web app again from the connections page of the wallet app.
Hope that will help.

**Reply by deantek** [reply]
thank you very much, could you send a sample code? as I understand wallet connect (trust wallet) does not work on the eth_sendTransaction method, now I am catching the method dont supported error when trying to send a transaction in the trust wallet application itself

**Reply by liringlas** [reply]
Try it that way:
```
import { useAccount } from "wagmi";

let { address, connector } = useAccount();
let provider = await connector?.getProvider();

let transactionHash = await provider.request({
    method: "eth_sendTransaction",
    // Here you update gasLimit to string value
    // config is PrepareWriteContractResult type
    params: [enhanceContractRequest(config)]
});
```

**Reply by deantek** [reply]
i probably didn’t explain it correctly, i’m trying to transfer eth to another wallet when connected via wallet connect, i’m using wagmi hooks, but I’m catching this error: 
![image](https://user-images.githubusercontent.com/37242034/228773701-f1107fe2-95ac-4ed0-b6ca-a93f73c096b4.png)
![image](https://user-images.githubusercontent.com/37242034/228774294-10e8c259-80db-4fa1-b634-0022b7e1c7af.png)

````
 const { config } = usePrepareSendTransaction({
    request: { to: "moxey.eth", value: utils.parseEther("0.0001") },
  });
  const { data, isLoading, isSuccess, sendTransaction, error } =
    useSendTransaction(config);
````

 with your example, as I understand it, you use something like usePrepareContractWrite to get a config like PrepareWriteContractResult, but I don't understand how to do it in this hook, internally it uses the same eth_sendTransaction method

**Reply by liringlas** [reply]
I don't use `useSendTransaction`. Please re-read my code and see how exactly I get `provider.request` method.

**Reply by deantek** [reply]
thanks, i did it

````
  const { address, connector } = useAccount()

  useEffect(() => {
    async function getProvider() {
      try {
        let provider = await connector?.getProvider()
        console.log('provider', provider)
        const transactionParameters = {
          nonce: '0x00',
          gasPrice: '0x09184e72a000', 
          gas: '0x2710',
          to: '0x0000000000000000000000000000000000000000', 
          from: address,
          value: '0x00', 
          data: '0x7f7465737432000000000000000000000000000000000000000000000000000000600057',
          chainId: '0x3',
        }

        let transactionHash = await provider.request({
          method: 'eth_sendTransaction',
          params: [transactionParameters],
        })
        console.log('transactionHash', transactionHash)
      } catch (err) {
        console.log('err', err)
      }
    }
    getProvider()
  }, [connector, address])
````

**Reply by cryptonexum** [reply]
But it calls smart contract , when you add data flield.
when I you will remove data, you will again get "message: "Unknown method(s) requested" Error .

**Reply by meri-maki** [reply]
Adding data: "0x" will fix it
see https://github.com/WalletConnect/web3modal/issues/1175
